src/components/Reports/GoalCalculator/Shared/useReportExpenses/helpers.test.tsx

```diff
@@ -0,0 +1,48 @@
+import { gqlMock } from '__tests__/util/graphqlMocking';
+import {} from 'src/graphql/types.generated';
+import {
+  BudgetFamilyFragment,
+  BudgetFamilyFragmentDoc,
+} from '../GoalCalculation.generated';
+import { getFamilyTotal, getPrimaryTotal } from './helpers';
+
+const mockFamily = gqlMock<BudgetFamilyFragment>(BudgetFamilyFragmentDoc, {
+  mocks: {
+    primaryBudgetCategories: [
+      {
+        directInput: 20,
+        subBudgetCategories: [{ amount: 100 }, { amount: 200 }],
+      },
+      {
+        directInput: null,
+        subBudgetCategories: [{ amount: 300 }, { amount: 400 }],
+      },
+    ],
+  },
+});
+
+describe('getPrimaryCategoryTotal', () => {
+  it('returns sum of all subcategory amounts for primary category with directInput set', () => {
+    expect(getPrimaryTotal(mockFamily.primaryBudgetCategories[0])).toBe(20);
+  });
+  it('returns sum of all subcategory amounts for primary category without directInput set', () => {
+    expect(getPrimaryTotal(mockFamily.primaryBudgetCategories[1])).toBe(700);
+  });
+});
+
+describe('getMinistryExpensesTotal', () => {
+  it('returns correct sum of all primary category amounts for ministry family with a directInput value set', () => {
+    expect(getFamilyTotal(mockFamily)).toBe(720);
+  });
+
+  it('returns sum of all primary category amounts for ministry family without any directInput set', () => {
```

Comment:
The test above is great! This one seems similar enough that I don't think it's strictly necessary. We do need a test case where the family itself has `directInput`.